File: src/main/java/ObjectsAndDataStructures/JavaCode/MainObject.java
package ObjectsAndDataStructures.JavaCode;

interface IStudentState {
    void working();

    void going();

    void relaxing();
}

public class MainObject {
    public static void main(String[] args) {
        IStudentState student = new IStudentState() {
            @Override
            public void working() {
                System.out.println("Student is working");
            }

            @Override
            public void going() {
                System.out.println("Student is going somewhere");
            }

            @Override
            public void relaxing() {
                System.out.println("Student is relaxing");
            }
        };

        student.working();
        student.going();
        student.relaxing();
    }
}

File: src/main/java/ObjectsAndDataStructures/cppCode/MainObject.cpp
#include <iostream>

using namespace std;

class StudentState {
public:
    virtual void working();

    virtual void going();

    virtual void relaxing();
};

class StudentStateRealization : public StudentState {
public:
    void working() {
        cout << "Student is working";
    }

    void going() {
        cout << "Student is going somewhere";
    }

    void relaxing() {
        cout << "Student is relaxing";
    }
};